compiler/optimizer/RedundantAsyncCheckRemoval.hpp

```diff
@@ -163,6 +152,27 @@ class TR_LoopEstimator
       };
    };
 
+
+/*
+ * Class TR_RedundantAsyncCheckRemoval
+ * ===================================
+ *
+ * Redundant async check removal optimization minimizes the number of async
+ * checks in a given loop. It uses an algorithm that attempts to achieve its 
+ * goal by adopting two distinct methods:
+ *
+ * 1. Check if the loop is known to be short running (either based on induction 
+ * variable information obtained from value propagation) or if the special 
+ * versioning test for short running loops is found. In either case, since 
+ * the loop is provably not long running and does not require an async check at all.
+ *
+ * 2. If async checks cannot be skipped completely for the loop, attempt to
+ * place the sync checks at points within the loop such that there is 
+ * minimal redundancy (two async checks along a control flow path are avoided 
+ * as far as possible). Note that non-INL calls are implicit async checks 
```

Comment:
INL methods are a java specific artifact that should move up at some point. Reference: http://www.ibm.com/support/knowledgecenter/SSYKE2_7.1.0/com.ibm.java.zos.71.doc/diag/tools/mt_untraceable.html